var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -342,6 +361,7 @@ class Trilinos(CMakePackage):
     patch('xlf_tpetra.patch', when='@12.12.1%xl')
     patch('xlf_tpetra.patch', when='@12.12.1%xl_r')
     patch('xlf_tpetra.patch', when='@12.12.1%clang')
+    patch('fix_clang_errors_12_18_1.patch', when='@12.18.1%clang')
```

Comment:
Have you explored which versions should this patch be applied to?

Reply:
> Should I limit the scope of the patch a bit more (i.e. only for my ancient compiler, apple-clang 8.0.0

Yes please (something like `when='12.8.1%clang@:8'` should work). Could you also mention what version of Mac OS you are running on?

Thanks!

Reply:
This patch seems pretty innocuous, and will likely be needed for other versions/compilers. I would go so far as to include it even for non-Clang compilers, but that's just me.

Reply:
@scheibelp I'm running 10.11 (El Capitan). Super old... Wouldn't `when='12.8.1%clang@:8'` also target clang built through spack, for example? Is there any way to restrict it to just apple clang, like `when='12.8.1%clang@:8 platform=darwin'` perhaps?

> This patch seems pretty innocuous

This is true. I think that it just needs a C++11 compliant compiler, but nowadays Trilinos requires C++11 to be built AFAIK. This is what I think I would have liked to have done, but I didn't want to appear overly zealous about patching.

Reply:
I see that there's a set of comments in the [cuda package](https://github.com/spack/spack/blob/4af630308619afb40b1de89d1f7bac0f57b70a38/lib/spack/spack/build_systems/cuda.py#L124-L132) that seem to reference the same idea of differentiating between Apple XCode clang and LLVM clang (more specifically, the difficulty in distinguishing which is the clang that's building this package on MacOS). What is the preferred way forward?